src/main/java/com/celements/webform/DocFormScriptService.java

```diff
@@ -83,6 +90,23 @@ private XWikiContext getContext() {
     }
     return docs;
   }
+
+  public Map<String, Set<DocumentReference>> updateAndSaveDocFromRequest() {
+    return updateAndSaveDocFromRequest(null);
+  }
+
+  @SuppressWarnings("unchecked")
+  public Map<String, Set<DocumentReference>> updateAndSaveDocFromRequest(
+      DocumentReference docRef) {
+    Collection<XWikiDocument> xdocs = Collections.<XWikiDocument>emptyList();
+    try {
+      xdocs = getDocFormCommand().updateDocFromMap(docRef,
+          getContext().getRequest().getParameterMap(), getContext());
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception in getDocFormCommand().updateDocFromMap()", xwe);
+    }
+    return getDocFormCommand().saveXWikiDocCollection(xdocs);
```

Comment:
First: Please check for edit-access rights before saving the xdocs. I suggest to save all or none (transaction behaviour).
Second: it does not make sense to me to save the documents if an exception occurred in preparation. Please move saveXWikiDocCollection inside the try block.